### PR TITLE
Migrate to substrate-4.0.0-dev

### PR DESCRIPTION
**Pull request by ivanceras**
## Related Issue Or Context

Closes: #103 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [X] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chainbridge/Cargo.toml [hunk 1]
```diff
@@ -1,42 +1,87 @@
+#TODO: rename to chainbridge pallet
 [package]
 name = 'chainbridge'
-version = '0.0.2'
-authors = ['david@chainsafe.io']
-edition = '2018'
+version = '0.0.3'
+authors = ['David Ansermino <david@chainsafe.io>', 'Jovansonlee Cesar <jovansonlee@chainsafe.io']
```
Comment (waymobetta) [inline chainbridge/Cargo.toml:5]: missing `>`

> `<jovansonlee@chainsafe.io']`

File: chainbridge/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+License: Unlicense
```
Comment (waymobetta) [inline chainbridge/README.md:1]: We may want to use `GNU Lesser General Public License v3.0`

This is the same that the repo was originally licensed under

**Comment by waymobetta** [comment]
Took a first pass and from a high level it looks good but I need to review further and try running tests (also looks like CI may have failed)

**Comment by ivanceras** [comment]
@willeslau, @clearloop hey guys, please spend sometime reviewing the migration of this chainbridge-pallet. I've thoroughly check everything here, but I need 1 more eye just to make sure.

**Pull request closed without merging**